### ISSUE TITLE
Input data for training and inference process

### ISSUE THREAD

**Issue by ShangWeize** [megvii-research/PETR#48]
Hi, I have the following 3 questions:
1. In PETRv2, the size of the img input in the training process is [12, 3, 320, 800]. Can 12 be understood as 6 pictures at the current moment and 6 pictures in the previous frame?
2. During the inference process, the input img size is [1, 12, 3, 320, 800]. Does the inference process input 12 pictures at two times at the same time?
3. When reading the dataloader, I did not find the process of adding the 6 pictures at the previous moment to the 6 pictures at the current moment. Which pipeline is this process reflected in?
thank you very much!

**Comment by yingfei1016** [comment]
Hi，
1、Yes, we concat previous 6 frames with current  6 frames at batch axis. Then it can be processed similar with single frame.
2、During the inference process, the previous frame is sampled at 15T. 
https://github.com/megvii-research/PETR/blob/main/projects/mmdet3d_plugin/datasets/pipelines/loading.py#L105
3、In the config we use LoadMultiViewImageFromMultiSweepsFiles to add previous frames  https://github.com/megvii-research/PETR/blob/main/projects/configs/petrv2/petrv2_vovnet_gridmask_p4_800x320.py#L183 .
In "LoadMultiViewImageFromMultiSweepsFiles" we add previous frames after line-121 , 
https://github.com/megvii-research/PETR/blob/main/projects/mmdet3d_plugin/datasets/pipelines/loading.py#L121

**Comment by ShangWeize** [comment]
thank you very much. I have one more question. Do you use the detection head and the segmentation head at the same time in petrv2_BEVseg.py? In addition, there are four weights of loss for these two tasks in petrv2_BEVseg.py. Is it enough to use the default weights?

**Comment by kvjia** [comment]
Hi. The default weights is enough to train both the detection head and the segmentation head at the same time. However, there is some thing wrong in the segmentation task, so we suggest you to train detection task firstly.

**Comment by ShangWeize** [comment]
Hi. What are the meanings of GioULoss and Sigmoid_ce_loss in the loss function?
